### PR TITLE
Add documentation and changelog for datetime bindings

### LINKED ISSUE #206: No documentation for datetime bindings
As part of #200, adding proper documentation to the Python datetime bindings was deferred to a later PR. This issue is to track the progress of adding proper rustdocs to the bindings.

To Do:

- [x] src/ffi/datetime.rs
- [x] src/objects/datetime.rs

The ffi bindings hew fairly closely to [the datetime C API](https://docs.python.org/3/c-api/datetime.html), so in general that can serve as a basis for the ffi documentation.

If someone else is interested in dipping their toes in the water, I'm happy to let someone else do the PR.

### PR DESCRIPTION
**Pull request by pganssle**
Fixes #206 and adds a changelog entry for this.

### CODE REVIEW

File: src/ffi/datetime.rs [hunk 1]
```diff
@@ -1,5 +1,9 @@
 #![cfg_attr(feature="cargo-clippy", allow(type_complexity))]
-
+/// FFI bindings to the functions and structs defined in `datetime.h`
+///
+/// This is the unsafe thin  wrapper around the [CPython C API](https://docs.python.org/3/c-api/datetime.html),
+/// and covers the various date and time related objects in the Python `datetime`
+/// standard library module.
```
Comment (konstin) [inline src/ffi/datetime.rs]: For module level docs, we need `//!` instead of `///`.

File: src/ffi/datetime.rs [hunk 2]
```diff
@@ -102,6 +106,7 @@ const _PyDateTime_DATETIME_DATASIZE: usize = 10;
 #[repr(C)]
 #[derive(Debug, Copy, Clone)]
 pub struct PyDateTime_Date {
+    /// Structure representing a `datetime.date`
```
Comment (konstin) [inline src/ffi/datetime.rs]: Unlike python, rust wants the doc strings outside the struct (The current would document `ob_base`)

**Comment by konstin** [comment]
Except for the noted technicalities, this looks great!

File: src/ffi/datetime.rs (hunk 2 above)
Comment (pganssle) [reply src/ffi/datetime.rs]: Yeah, I noticed this about halfway through, then went back to fix them. Must have missed at least this one.

**Comment by konstin** [comment]
Thank you!

**Pull request merged**